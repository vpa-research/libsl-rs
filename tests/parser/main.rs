use std::fs;

use insta::{assert_ron_snapshot, Settings};
use libsl::{LibSl, WithLibSl};
use pathdiff::diff_paths;

#[test]
fn test_snapshots() {
    let mut settings = Settings::clone_current();
    settings.set_omit_expression(true);
    let _guard = settings.bind_to_scope();

    insta::glob!("test-data/**/*.libsl", |path| {
        let contents = fs::read_to_string(path)
            .map_err(|e| format!("could not read `{}` to string: {e}", path.display()))
            .unwrap();

        let mut settings = Settings::clone_current();
        settings.set_description(format!("// Input file:\n{contents}"));
        let _guard = settings.bind_to_scope();

        let mut libsl = LibSl::new();
        let ast = libsl.parse_file(
            diff_paths(path, env!("CARGO_MANIFEST_DIR")).unwrap().display().to_string(),
            &contents
        );

        assert_ron_snapshot!(match &ast {
            Ok(file) => Ok(file.with_libsl(&libsl)),
            Err(e) => Err(e.to_string()),
        });
    });
}
