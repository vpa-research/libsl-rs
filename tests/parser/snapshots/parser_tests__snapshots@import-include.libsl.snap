---
source: tests/parser/main.rs
description: "// Input file:\nlibsl \"1.0.0\";\n\nlibrary test_imp_incl;\n\nimport hello_world;\ninclude foo_bar;\n"
input_file: tests/parser/test-data/import-include.libsl
---
Ok(File(
  loc: Span(
    start: 0,
    len: 76,
    file: "tests/parser/test-data/import-include.libsl",
    line: Some(1),
    col: None,
  ),
  header: Some(Header(
    loc: Span(
      start: 0,
      len: 37,
      file: "tests/parser/test-data/import-include.libsl",
      line: Some(1),
      col: None,
    ),
    libsl_version: "1.0.0",
    library_name: "test_imp_incl",
    version: None,
    language: None,
    url: None,
  )),
  decls: [
    Decl(
      loc: Span(
        start: 40,
        len: 18,
        file: "tests/parser/test-data/import-include.libsl",
        line: Some(5),
        col: None,
      ),
      kind: Import(DeclImport(
        path: "hello_world",
      )),
    ),
    Decl(
      loc: Span(
        start: 60,
        len: 15,
        file: "tests/parser/test-data/import-include.libsl",
        line: Some(6),
        col: None,
      ),
      kind: Include(DeclInclude(
        path: "foo_bar",
      )),
    ),
  ],
))
